trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestHelper.java

```diff
@@ -0,0 +1,298 @@
+package io.bitrise.trace.plugin.util;
+
+import androidx.annotation.NonNull;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.junit.rules.TestName;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import io.bitrise.trace.plugin.TraceConstants;
+
+/**
+ * Helper class to support {@link io.bitrise.trace.plugin.TraceGradlePluginFunctionalTest}
+ */
+public class FunctionalTestHelper {
+
+    /**
+     * A temporary folder for running functional tests. A new gradle project will be setup in this folder, and the
+     * tests will run in this project.
+     */
+    private static final File testProjectDir =
+            new File(System.getProperty("user.home") + "/BitriseTrace-TestGradleProject");
+
+    @NonNull
+    private final Logger logger;
+
+    public FunctionalTestHelper() {
+        this.logger = Logging.getLogger(this.getClass().getName());
+    }
+
+    /**
+     * Publishes the trace-gradle-plugin itself and logs it.
+     */
+    public void publishTraceGradlePlugin() {
+        logger.lifecycle("Publishing plugin, to make sure it is up-to-date");
+        FunctionalTestUtils.publishTraceGradlePlugin();
+        logger.lifecycle("Publishing plugin finished");
+    }
+
+    /**
+     * Prints multiple lines with the given text in a pretty ASCII text box.
+     *
+     * @param text the text to print.
+     */
+    public void logInAsciiBox(@NonNull final String text) {
+        final int textLength = text.length();
+        final int paddingLength = 30;
```

Comment:
what is this padding length doing? If i understand correctly, does this mean that if the text i pass in is super long, the box widths will change? Would it be worth considering a max line length instead? so that the boxes are uniform in length and if we get one super long text, it'll automatically wrap the text? This would mean the padding might change between the borders, but maybe a min padding would be useful e.g. 5 chars either side?

Reply:
The padding length is the empty space around the test name. There is definitely a way to improve the behaviour, I decided to keep it simple because:
- this is used only for tests, no additonal functionality
- unlikely to have a super long test name (I mean it would cry for a refactor/rename if we had): let's say we would cap line length at 120 chars, it would mean ~90 chars for the testname
- if worst case scenario happens, and we have a super long name, it's functionality will be still provided, the only thing will happen it will be less good looking

Of course please let me know if I missed anything, or you would strongly prefer doing these improvements.

Reply:
Hmm - sure. I think that sounds fair :) Potentially we could have a tech debt ticket to refactor and improve this in the future?

Reply:
Sure, created a ticket for it.